BAUF-106 zaslon o poslovima na koje radnik eka

dmatijani opened the pull request:
**Napravljen je zaslon za poslove na koje korisnik čeka (odobrenje ili prihvaćanje):**
![image](https://github.com/user-attachments/assets/ea7f9fd0-352b-44cb-ab17-a81b95dfbb94)

Dodana je i ikonica "My jobs" na navbaru prema skicama.

Pratio sam skicu na sljedećoj poveznici: https://vlovric21.atlassian.net/wiki/spaces/baufind/pages/9568439/Kao+korisnik+elim+biti+obavije+ten+o+promijenama+vezanih+uz+svoje+zahtjeve+kako+bih+znao+status+odvijanja+zahtjeva

Napravio sam novi API poziv na _/jobs/searchMyJobsForCurrentUser_, i vraćaju se poslovi, kao i nazivi statusa:
![image](https://github.com/user-attachments/assets/dea7032a-cd28-4cfe-925d-bdaa0974cd57)

Što se tiče koda, uglavnom sam se držao strukture kao i do sada, na backendu možda ima malo ponavljanja (jer postoji putanja _CheckPendingInvitations_), no ona ne vraća statuse, samo poslove, a nisam htio nju mijenjati, iako mislim da nije toliko bitno jer je rečeno za drugu provjeru da je bitno da backend radi svoj posao, a pregledavanje koda će se uglavnom raditi na frontendu.
Na frontendu sam se držao strukture, no ako mislite da je nešto krivo, ili na krivom mjestu, samo recite i popravit ću.

Još dvije stvari:
1) Napravio sam samo zaslon s popisom, bez da se može otvoriti posao i bez da se može kliknuti na ikonicu povijesti posla, **da napravim i to? Ako da, recite pa ću odmah na ovom PR-u napraviti.**
2) Napravio sam da se na karticama posla za status ispišu vrijednosti za taj status iz baze (tablica _working status_):
![image](https://github.com/user-attachments/assets/b1d9324d-2160-4ac2-8e5b-5248f64ff288)
**Da to promijenim u bazi (nazive), ili da na backendu prema ID-u nekako ja napišem status ručno?**

dmatijani committed:
BAUF-106 Kreiran prazan zaslon mojih poslova kao i pripadajući ViewModel

Još se ništa ne prikazuje, osim teksta, a i ViewModel klasa ništa ne sadrži.

dmatijani committed:
BAUF-106 Backend logika za dohvat poslova za korisnika

Poslovi koji nisu gotovi. Ti će ići u povijest poslova, ali će se dio implementirane logike moći ponovno iskoristiti.

dmatijani committed:
BAUF-106 Moji poslovi se učitavaju na frontendu

Još ih treba prikazati.

dmatijani committed:
BAUF-106 Zaslon za poslove na koje korisnik čeka

fsimic21 reviewed (changes requested):
Kao što je kolega Viktor rekao (Whatsapp) , treba napraviti te promjene pa opet poslati.

vlovric21 reviewed (changes requested):
Sve je uredno napravljeno ali je krivo shvaćena svrha ekrana. Ovaj ekran treba sadržavati popis poslova čiji je vlasnik korisnik, a da poslovi ili nisu još počeli sa radom ili su u tijeku. Znači ukratko trenutno postojeći poslovi za tog korisnika.
No sad dok promatram wireframe zapravo mi se čini da nemamo ekran na kojemu radnik može vidjeti poslove koje obavlja tj. na koje je prihvaće, tako da možda da to ide na ovaj ekran? Trebalo bi se dogovorit

dmatijani commented:
> Sve je uredno napravljeno ali je krivo shvaćena svrha ekrana. Ovaj ekran treba sadržavati popis poslova čiji je vlasnik korisnik, a da poslovi ili nisu još počeli sa radom ili su u tijeku. Znači ukratko trenutno postojeći poslovi za tog korisnika. No sad dok promatram wireframe zapravo mi se čini da nemamo ekran na kojemu radnik može vidjeti poslove koje obavlja tj. na koje je prihvaće, tako da možda da to ide na ovaj ekran? Trebalo bi se dogovorit

U redu, popravit ću to, nakon sastanka kad vidimo što ćemo s ovim ekranom, jer mislim da je i on potreban.

dmatijani committed:
Merge branch 'develop' into BAUF-106-Zaslon-o-poslovima-na-koje-radnik-eka

dmatijani committed:
BAUF-106 Zaslon za My Jobs na frontendu pretvoren u Pending Jobs

Zašto: https://github.com/AIR-DavidoviTigrovi/baufind/pull/39
Što: https://vlovric21.atlassian.net/wiki/spaces/baufind/pages/43941890/07.01.2025.

dmatijani committed:
BAUF-106 Refaktorirani nazivi klasa

Uglavnom iz "MyJobs..." u "PendingJobs..."

dmatijani committed:
BAUF-106 Dodan novi zaslon "Moji poslovi"

Zaslon sada prikazuje ono što treba.

dmatijani commented:
@vlovric21 @fsimic21 Ispravio sam sve što je bilo krivo i sada su ovo novi zasloni:

**Zaslon Moji poslovi (1/2):**
![moji poslovi 1](https://github.com/user-attachments/assets/602612c3-ed93-4e69-af7e-8e7d051fa16f)

**Zaslon Moji poslovi (2/2):**
![moji poslovi 2](https://github.com/user-attachments/assets/ffcea432-ca24-47f9-a57f-01fcd5d23c84)

**Gumb na profilu:**
![profil](https://github.com/user-attachments/assets/7d09560a-bb4a-4ad5-9a22-c984ce50bc65)

**Novoimenovani zaslon Pending jobs:**
![pending jobs](https://github.com/user-attachments/assets/4e548cac-c290-492b-9369-0d78dffb3250)

Ako što treba promijeniti, bilo to poziciju gumba na profilu ili nazive ili nešto, recite. Miješan je hrvatski i engleski, da, ali to budemo riješili u 5. sprintu.

dmatijani committed:
BAUF-106 Dodana provjera statusa

Radnik baš mora biti primljen na posao da se vidi u "My jobs".

vlovric21 reviewed (approved):
Super, sada je sve kako je zamišljeno. Pohvale na UI-u kako si odvojio na My Jobs kategorije poslova. Sve uredno.

fsimic21 reviewed (approved):
Sad izgleda sve kako treba, to bi bilo to. Svaka čast na čistoći koda.
